tools/scripts/travis_tests.sh

```diff
@@ -31,7 +28,12 @@ if [[ -n "$illegal_commits" ]]; then
     exit 1
 fi;
 
-source init_env
+# Some commands are allowed to fail in init_env, e.g. to probe for installed
+# tools. However, the overall script has to succeed.
+source init_env || exit 1
```

Comment:
Shouldn't we make sure sourcing `init_env` always work? I suppose if it breaks the rest of the script will fail, but then it'll be less obvious where the failure came from.

Reply:
That's exactly why I use `|| exit 1`: if init_env fails, then travis_tests.sh will fail as well.
We just do not fail at every command that fails inside init_env. Some of them are expected to fail under normal circumstances since they probe for existing things (like jupyter being installed).